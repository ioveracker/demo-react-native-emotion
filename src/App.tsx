import styled from '@emotion/native';
import { ThemeProvider } from '@emotion/react';
import React, { useState } from 'react';
import { SafeAreaView, StatusBar, Switch } from 'react-native';
import { darkTheme, lightTheme } from './themes';

const App = () => {
    const [theme, setTheme] = useState(lightTheme);
    return (
        <ThemeProvider theme={theme}>
            <StatusBar
                barStyle={
                    theme.name === 'dark' ? 'light-content' : 'dark-content'
                }
            />
            <SafeAreaContainer>
                <ScrollView contentInsetAdjustmentBehavior="automatic">
                    <SectionContainer>
                        <ThemeSwitchContainer>
                            <SectionTitle>Dark Mode</SectionTitle>
                            <Switch
                                value={theme.name === 'dark'}
                                onValueChange={(value) => {
                                    if (value) {
                                        setTheme(darkTheme);
                                    } else {
                                        setTheme(lightTheme);
                                    }
                                }}
                            />
                        </ThemeSwitchContainer>
                    </SectionContainer>
                    <SectionContainer>
                        <SectionTitle>Step One</SectionTitle>
                        <SectionDescription>
                            Edit <HighlightedText>App.tsx</HighlightedText> to
                            change this screen and then come back to see your
                            edits.
                        </SectionDescription>
                    </SectionContainer>
                    <SectionContainer>
                        <SectionTitle>See Your Changes</SectionTitle>
                        <SectionDescription>
                            Press Cmd + R in the simulator to reload your app's
                            code.
                        </SectionDescription>
                    </SectionContainer>
                    <SectionContainer>
                        <SectionTitle>Debug</SectionTitle>
                        <SectionDescription>
                            Press Cmd + D in the simulator or shake your device
                            to open the React Native debug menu.
                        </SectionDescription>
                    </SectionContainer>
                    <SectionContainer>
                        <SectionTitle>Learn More</SectionTitle>
                        <SectionDescription>
                            Read the docs to discover what to do next.
                        </SectionDescription>
                    </SectionContainer>
                </ScrollView>
            </SafeAreaContainer>
        </ThemeProvider>
    );
};

const SafeAreaContainer = styled(SafeAreaView)`
    background-color: ${(props) => props.theme.colors.background};
    flex-grow: 1;
`;

const ScrollView = styled.ScrollView`
    background-color: ${(props) => props.theme.colors.background};
`;

const SectionContainer = styled.View`
    margin-top: 32px;
    padding: 0px 24px;
`;

const ThemeSwitchContainer = styled.View`
    flex-direction: row;
    justify-content: space-between;
`;

const SectionTitle = styled.Text`
    font-size: 24px;
    font-weight: 600;
    color: ${(props) => props.theme.colors.titleFont};
`;

const SectionDescription = styled.Text`
    margin-top: 8px;
    font-size: 18px;
    font-weight: 400;
    color: ${(props) => props.theme.colors.descriptionFont};
`;

const HighlightedText = styled.Text`
    font-weight: 700;
`;

export default App;
