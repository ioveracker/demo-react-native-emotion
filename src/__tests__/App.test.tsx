import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    renderer.create(<App />);
});

it('matches snapshot', () => {
    expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
});
