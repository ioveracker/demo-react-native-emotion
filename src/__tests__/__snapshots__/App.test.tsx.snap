// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexGrow": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "marginTop": 32,
              "paddingBottom": 0,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#222",
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                undefined,
              ]
            }
          >
            Dark Mode
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 32,
              "paddingBottom": 0,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 24,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Step One
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#666",
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              },
              undefined,
            ]
          }
        >
          Edit 
          <Text
            style={
              Array [
                Object {
                  "fontWeight": "700",
                },
                undefined,
              ]
            }
          >
            App.tsx
          </Text>
           to change this screen and then come back to see your edits.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 32,
              "paddingBottom": 0,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 24,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          See Your Changes
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#666",
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              },
              undefined,
            ]
          }
        >
          Press Cmd + R in the simulator to reload your app's code.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 32,
              "paddingBottom": 0,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 24,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Debug
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#666",
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              },
              undefined,
            ]
          }
        >
          Press Cmd + D in the simulator or shake your device to open the React Native debug menu.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 32,
              "paddingBottom": 0,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 24,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Learn More
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#666",
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              },
              undefined,
            ]
          }
        >
          Read the docs to discover what to do next.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
